packages/js/product-editor/src/hooks/use-conditional-execution.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { RefObject } from 'react';
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useVisibilityObserver } from './use-visibility-observer';
+
+type useConditionalExecutionProps = {
+	elementOrSelector?: RefObject< HTMLInputElement > | string;
```

Comment:
If the ref is passed through the props then why having the one defined in line 25? If we receive it as a prop then we don't need to return it as in line 51.

Reply:
Good call! I fixed it in the commit [a8cc850](https://github.com/woocommerce/woocommerce/pull/47025/commits/a8cc85072cedf02efcef288e5aac2ef8ae141689)